### ISSUE TITLE
Tests fail with cudf 0.7.1

### ISSUE THREAD

**Issue by randyzwitch** [omnisci/pymapd#236]
When running the test suite with cudf 0.7.1, getting failures about read-only buffers

```python
=================================== FAILURES ===================================
_________________ TestGPUDataNoNulls.test_select_ipc_gpu[rows] _________________

self = <tests.test_data_no_nulls_gpu.TestGPUDataNoNulls object at 0x7fa5550b8c18>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
method = 'rows'

    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    @pytest.mark.parametrize('method', ["rows", "columnar", "arrow", "infer"])
    def test_select_ipc_gpu(self, con, method):
    
        df_in = _tests_table_no_nulls(10000)
        # keep columns that pass, make issues for ones that don't
        df_in_limited = df_in[["smallint_",
                               "int_",
                               "bigint_",
                               "float_",
                               "double_"]].copy()
    
        con.execute("drop table if exists test_data_no_nulls_gpu;")
        con.load_table("test_data_no_nulls_gpu", df_in_limited, method=method)
>       df_gdf = con.select_ipc_gpu("select * from test_data_no_nulls_gpu")

tests/test_data_no_nulls_gpu.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
_______________ TestGPUDataNoNulls.test_select_ipc_gpu[columnar] _______________

self = <tests.test_data_no_nulls_gpu.TestGPUDataNoNulls object at 0x7fa4fe581a20>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
method = 'columnar'

    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    @pytest.mark.parametrize('method', ["rows", "columnar", "arrow", "infer"])
    def test_select_ipc_gpu(self, con, method):
    
        df_in = _tests_table_no_nulls(10000)
        # keep columns that pass, make issues for ones that don't
        df_in_limited = df_in[["smallint_",
                               "int_",
                               "bigint_",
                               "float_",
                               "double_"]].copy()
    
        con.execute("drop table if exists test_data_no_nulls_gpu;")
        con.load_table("test_data_no_nulls_gpu", df_in_limited, method=method)
>       df_gdf = con.select_ipc_gpu("select * from test_data_no_nulls_gpu")

tests/test_data_no_nulls_gpu.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
________________ TestGPUDataNoNulls.test_select_ipc_gpu[arrow] _________________

self = <tests.test_data_no_nulls_gpu.TestGPUDataNoNulls object at 0x7fa4fee8b1d0>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
method = 'arrow'

    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    @pytest.mark.parametrize('method', ["rows", "columnar", "arrow", "infer"])
    def test_select_ipc_gpu(self, con, method):
    
        df_in = _tests_table_no_nulls(10000)
        # keep columns that pass, make issues for ones that don't
        df_in_limited = df_in[["smallint_",
                               "int_",
                               "bigint_",
                               "float_",
                               "double_"]].copy()
    
        con.execute("drop table if exists test_data_no_nulls_gpu;")
        con.load_table("test_data_no_nulls_gpu", df_in_limited, method=method)
>       df_gdf = con.select_ipc_gpu("select * from test_data_no_nulls_gpu")

tests/test_data_no_nulls_gpu.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
________________ TestGPUDataNoNulls.test_select_ipc_gpu[infer] _________________

self = <tests.test_data_no_nulls_gpu.TestGPUDataNoNulls object at 0x7fa4fe7499b0>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
method = 'infer'

    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    @pytest.mark.parametrize('method', ["rows", "columnar", "arrow", "infer"])
    def test_select_ipc_gpu(self, con, method):
    
        df_in = _tests_table_no_nulls(10000)
        # keep columns that pass, make issues for ones that don't
        df_in_limited = df_in[["smallint_",
                               "int_",
                               "bigint_",
                               "float_",
                               "double_"]].copy()
    
        con.execute("drop table if exists test_data_no_nulls_gpu;")
        con.load_table("test_data_no_nulls_gpu", df_in_limited, method=method)
>       df_gdf = con.select_ipc_gpu("select * from test_data_no_nulls_gpu")

tests/test_data_no_nulls_gpu.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
___ TestIntegration.test_select_ipc_gpu[select qty, price from stocks-None] ____

self = <tests.test_integration.TestIntegration object at 0x7fa4fe770358>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
query = 'select qty, price from stocks', parameters = None

    @pytest.mark.parametrize('query, parameters', [
        ('select qty, price from stocks', None),
        ('select qty, price from stocks where qty=:qty', {'qty': 100}),
    ])
    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    def test_select_ipc_gpu(self, con, query, parameters):
    
        from cudf.dataframe import DataFrame
    
        c = con.cursor()
        c.execute('drop table if exists stocks;')
        create = ('create table stocks (date_ text, trans text, symbol text, '
                  'qty int, price float, vol float);')
        c.execute(create)
        i1 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14,1.1);"  # noqa
        i2 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','GOOG',100,12.14,1.2);"  # noqa
    
        c.execute(i1)
        c.execute(i2)
    
>       result = con.select_ipc_gpu("select qty, price from stocks")

tests/test_integration.py:230: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
_ TestIntegration.test_select_ipc_gpu[select qty, price from stocks where qty=:qty-parameters1] _

self = <tests.test_integration.TestIntegration object at 0x7fa4fe314828>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)
query = 'select qty, price from stocks where qty=:qty'
parameters = {'qty': 100}

    @pytest.mark.parametrize('query, parameters', [
        ('select qty, price from stocks', None),
        ('select qty, price from stocks where qty=:qty', {'qty': 100}),
    ])
    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    def test_select_ipc_gpu(self, con, query, parameters):
    
        from cudf.dataframe import DataFrame
    
        c = con.cursor()
        c.execute('drop table if exists stocks;')
        create = ('create table stocks (date_ text, trans text, symbol text, '
                  'qty int, price float, vol float);')
        c.execute(create)
        i1 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14,1.1);"  # noqa
        i2 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','GOOG',100,12.14,1.2);"  # noqa
    
        c.execute(i1)
        c.execute(i2)
    
>       result = con.select_ipc_gpu("select qty, price from stocks")

tests/test_integration.py:230: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
___________________ TestIntegration.test_select_gpu_first_n ____________________

self = <tests.test_integration.TestIntegration object at 0x7fa4fe29c908>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)

    @pytest.mark.skipif(no_gpu(), reason="No GPU available")
    def test_select_gpu_first_n(self, con):
    
        c = con.cursor()
        c.execute('drop table if exists stocks;')
        create = ('create table stocks (date_ text, trans text, symbol text, '
                  'qty int, price float, vol float);')
        c.execute(create)
        i1 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14,1.1);"  # noqa
        i2 = "INSERT INTO stocks VALUES ('2006-01-05','BUY','GOOG',100,12.14,1.2);"  # noqa
    
        c.execute(i1)
        c.execute(i2)
    
>       result = con.select_ipc_gpu("select * from stocks", first_n=1)

tests/test_integration.py:255: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pymapd/connection.py:306: in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)
pymapd/_parsers.py:203: in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)
cudf/bindings/gpuarrow.pyx:203: in cudf.bindings.gpuarrow.GpuArrowReader.__init__
    ???
cudf/bindings/gpuarrow.pyx:228: in cudf.bindings.gpuarrow.GpuArrowReader._open
    ???
cudf/bindings/gpuarrow.pyx:248: in cudf.bindings.gpuarrow.GpuArrowReader._parse_metdata
    ???
stringsource:654: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: buffer source array is read-only

stringsource:349: ValueError
_______________ TestIntegration.test_dashboard_duplication_remap _______________

self = <tests.test_integration.TestIntegration object at 0x7fa4fe27e208>
con = Connection(mapd://mapd:***@localhost:6274/mapd?protocol=binary)

    def test_dashboard_duplication_remap(self, con):
        # This test relies on the test_data_no_nulls_ipc table
        # Setup our testing variables
        old_dashboard_state = dashboard_metadata.old_dashboard_state
        old_dashboard_name = dashboard_metadata.old_dashboard_name
        new_dashboard_name = "new_test"
        meta_data = {"table": "test_data_no_nulls_ipc", "version": "v2"}
        remap = {
                    "test_data_no_nulls_ipc": {
                        "name": new_dashboard_name,
                        "title": new_dashboard_name
                        }
                }
        dashboard_id = ""
    
        # Create testing dashboard
        con._client.create_dashboard(
            session=con._session,
            dashboard_name=old_dashboard_name,
            dashboard_state=(
                base64.b64encode(json.dumps(old_dashboard_state).encode(
                    "utf-8"))),
            image_hash="",
>           dashboard_metadata=json.dumps(meta_data),
        )

tests/test_integration.py:350: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mapd/MapD.py:2167: in create_dashboard
    return self.recv_create_dashboard()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <mapd.MapD.Client object at 0x7fa55503ef60>

    def recv_create_dashboard(self):
        iprot = self._iprot
        (fname, mtype, rseqid) = iprot.readMessageBegin()
        if mtype == TMessageType.EXCEPTION:
            x = TApplicationException()
            x.read(iprot)
            iprot.readMessageEnd()
            raise x
        result = create_dashboard_result()
        result.read(iprot)
        iprot.readMessageEnd()
        if result.success is not None:
            return result.success
        if result.e is not None:
>           raise result.e
E           mapd.ttypes.TMapDException: TMapDException(error_msg='Dashboard with name: old_dashboard already exists.')

mapd/MapD.py:2195: TMapDException
=========================== short test summary info ============================
SKIPPED [1] tests/test_deallocate.py:34: deallocate non-functional in recent distros
SKIPPED [1] tests/test_deallocate.py:54: deallocate non-functional in recent distros
SKIPPED [1] tests/test_deallocate.py:67: deallocate non-functional in recent distros
SKIPPED [1] tests/test_deallocate.py:80: deallocate non-functional in recent distros
SKIPPED [1] tests/test_deallocate.py:92: deallocate non-functional in recent distros
SKIPPED [1] tests/test_deallocate.py:105: deallocate non-functional in recent distros
=============== 8 failed, 71 passed, 6 skipped in 28.70 seconds ================
```

**Comment by randyzwitch** [comment]
This also showing an error that dashboard exists (when not using ephemeral container), will fix in separate PR.

**Comment by randyzwitch** [comment]
Ref: https://github.com/rapidsai/cudf/issues/1766

**Comment by randyzwitch** [comment]
Closed by cudf 0.7.2 release
